components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/RecoverPasswordApiServiceImpl.java

```diff
@@ -87,13 +87,13 @@ public Response recoverPasswordPost(RecoveryInitiatingRequestDTO recoveryInitiat
             }
             RecoveryUtil.handleBadRequest(e.getMessage(), e.getErrorCode());
         } catch (IdentityRecoveryException e) {
+            if (e.getCause() != null && StringUtils.equals(Constants.ERROR_MESSAGE_EMAIL_NOT_FOUND,
+                    e.getCause().getMessage())) {
+                LOG.error(e.getCause().getMessage(), e.getCause());
+                RecoveryUtil.handleBadRequest(e.getCause().getMessage(), Constants.ERROR_CODE_EMAIL_NOT_FOUND);
+            }
             RecoveryUtil.handleInternalServerError(Constants.SERVER_ERROR, e.getErrorCode(), LOG, e);
         } catch (Throwable throwable) {
-            if (throwable != null && StringUtils.equals(Constants.ERROR_MESSAGE_EMAIL_NOT_FOUND,
```

Comment:
Isn't there any place we are throwing this error in a different exception? not IdentityRecoveryException?

Reply:
The error message in throwable should be equal to"Email notification sending failed. Sending email address is not configured for the user." [Constants.ERROR_MESSAGE_EMAIL_NOT_FOUND] to go into if block. Searched in org:wso2 and org:wso2-extensions and the exact error message usage is only found at one place which thrown as a IdentityRecoveryException [https://github.com/wso2-extensions/identity-event-handler-notification/blob/2af7975d7a56a62f079842cc8bbb58a3cfc8697a/components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/util/NotificationUtil.java#L322].

Reply:
Just check more on this. We can't bind with the error message because there may be other places without checking on the error message we may use the exception from [1], so there can be regressions so we have to keep throwable part as it is. 

[1] https://github.com/wso2-extensions/identity-event-handler-notification/blob/2af7975d7a56a62f079842cc8bbb58a3cfc8697a/components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/util/NotificationUtil.java#L322